src/func/imports/recent-imports.js

```diff
@@ -0,0 +1,114 @@
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+// @flow
+
+import type {EntityTypeString, Transaction} from '../types';
+import _ from 'lodash';
+import {entityTypes} from '../entity';
+import {getAliasByIds} from '../alias';
+import moment from 'moment';
+
+/** getRecentImportIdsByType -
+ * @param  {Transaction} transacting - Transaction object
+ * @param  {number} limit - To limit the number of importIds fetched
+ * @param  {number} offset - We offset the result by this value
+ * @returns {Promise<Object<Array>>} - Returns an object containing arrays of
+ * 		importIds of various importTypes
+ */
+async function getRecentImportIdsByType(
+	transacting: Object,
+	limit: number,
+	offset: number
+) {
+	// Extract recent imports, types and import timeStamp
+	const recentImportIdAndType: Object =
+			await transacting.select(
+				'link.imported_at',
+				'bookbrainz.import.id',
+				'bookbrainz.import.type'
+			)
+				.from('bookbrainz.import')
+				.join(
+					transacting.select('import_id', 'imported_at')
+						.from('bookbrainz.link_import')
+						.orderBy('imported_at')
+						.limit(limit)
+						.offset(offset)
+						.as('link'),
+					'link.import_id',
+					'bookbrainz.import.id'
+				);
+
+	/* Contruct importHolder object (holds importIds classified by their types)
+		object of form {type: []} */
+	const importHolder = _.values(entityTypes).reduce(
+		(holder: Object, type: string) => _.assign(holder, {[type]: []}), {}
+	);
+
+	/* Returns the holder object which has two fields:
+		=> importHolder: Object{type: Array[importIds]}
+		=> timestampMap: Object{importId: imported_at} */
+	return recentImportIdAndType.reduce((holder: Object, idType: Object) => {
+		holder.importHolder[idType.type].push(idType.id);
+		holder.timeStampMap[idType.id] = idType.imported_at;
+		return holder;
+	}, {importHolder, timeStampMap: {}});
+}
+
+function getRecentImportsByType(transacting, type, importIds) {
+	return transacting.select('*')
+		.from(`bookbrainz.${type.toLowerCase()}_import`)
+		.whereIn('import_id', importIds);
+}
+
+export default async function getRecentImports(
+	orm: Object, transacting: Transaction, limit: number = 10,
+	offset: number = 0
+) {
+	// Fetch most recent ImportIds classified by importTypes
+	const {importHolder: recentImportIdsByType, timeStampMap} =
+		await getRecentImportIdsByType(transacting, limit, offset);
+
+	const importTypes: Array<EntityTypeString> = _.values(entityTypes);
+	// Fetch imports for each importType using their importIds
+	const recentImports = _.flatten(
+		await Promise.all(
+			importTypes.map(type =>
+				getRecentImportsByType(transacting, type,
+					recentImportIdsByType[type]))
+		)
+	);
+
+	const defaultAliasIds = _.map(recentImports, 'default_alias_id');
+	const defaultAliases =
+		await getAliasByIds(transacting, defaultAliasIds);
+	// Append defaultAlias objects to the recentImports
+	defaultAliases.forEach((defaultAlias, index) => {
```

Comment:
Can probably merge this into the loop below, if you return a map of aliasId, alias pairs, then just assign `recentImport.defaultAlias = defaultAliases[recentImport.defaultAliasId]`.

Reply:
Right, should have paid more attention.